File: src/WebApp/src/app/Models/ResourceDto.ts
export class ResourceDto {

  constructor(
    private resourceID: number,
    private name: string,
    private category: number,
    private isExhaustable: boolean,
    private isAvailable: number,
    private description: string,
    private supplierID: number,
    private quantity: number,
  ) {}

  toCreateRequestBody() {
  return {
    // Don't include resourceID here
    name: this.name,
    category: this.category,
    isExhaustable: this.isExhaustable,
    isAvailable: this.isAvailable,
    description: this.description,
    supplierID: this.supplierID,
    quantity: this.quantity
  };
}


  getResourceID(): number {
    return this.resourceID;
  }

  getName(): string {
    return this.name;
  }

  getCategory(): number {
    return this.category;
  }

  getIsExhaustable(): boolean {
    return this.isExhaustable;
  }

  getIsAvailable(): number {
    return this.isAvailable;
  }

  getDescription(): string {
    return this.description;
  }

  getSupplierID(): number {
    return this.supplierID;
  }

  getQuantity(): number {
    return this.quantity;
  }

}

File: src/Backend/Backend/Models/Dto/ResourceDto.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Backend.Models.Dto
{
    public class ResourceDto
    {
        public int ResourceID { get; set; }
        public string Name { get; set; }
        public ResourceCategory Category { get; set; }
        public bool IsExhaustable { get; set; }
        public ResourceAvailability IsAvailable { get; set; }
        public string Description { get; set; }
        public int SupplierID { get; set; }
        public int Quantity { get; set; }
        public User Supplier { get; set; }
    }
}
